portfolio/src/main/webapp/game.js

```diff
@@ -0,0 +1,82 @@
+
+
+
+var myVar, canvas, ctx, x, y, dx, dy, factor, color, colorStr;
```

Comment:
Global vars are discouraged in js. Can you make sure among these vars, only the one you absolutely need to be global are declared here?